src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -329,7 +330,8 @@ internal IEnumerable<DiagnosticAnalyzer> GetAnalyzers(Project project)
                 var versions = new VersionArgument(projectTextVersion, semanticVersion, projectVersion);
                 foreach (var stateSet in _stateManager.GetOrUpdateStateSets(project))
                 {
-                    if (SkipRunningAnalyzer(project.CompilationOptions, analyzerDriver, openedDocument: false, skipClosedFileChecks: false, stateSet: stateSet))
+                    // Compilation actions can report diagnostics on open files, so we skipClosedFileChecks.
+                    if (SkipRunningAnalyzer(project.CompilationOptions, analyzerDriver, openedDocument: true, skipClosedFileChecks: true, stateSet: stateSet))
                     {
```

Comment:
You mean if compilation end action reports diagnostics on closed files? Current implementation doesn't, but I am not sure whether it should or should not. Its a valid diagnostic within the project, so why not show it to the user if it already been computed? Probably warrants creating a separate issue for discussion?